packages/react-components/react-tree/src/components/TreeItem/useTreeItem.tsx

```diff
@@ -131,6 +134,10 @@ export const useTreeItem_unstable = (props: TreeItemProps, ref: React.Ref<HTMLDi
       role: 'treeitem',
       onClick: handleClick,
       onKeyDown: handleKeyDown,
+      onMouseOver: actions ? showActions : undefined,
+      onFocus: actions ? showActions : undefined,
+      onMouseOut: actions ? hideActions : undefined,
+      onBlur: actions ? hideActions : undefined,
```

Comment:
Depending on event listeners instead of pure CSS allows us to provide the visibility of actions through context, decoupling the logic of what should be visible from `TreeItem` and allowing modifications on its children.

The downside is that depending on simple CSS styles to hide and present actions would be better for performance.

Reply:
Another alternative would be to use CSS variables... 🤔

Reply:
to set css variables you'd still need to re-render no?

Reply:
No! I'd be setting them over `:hover` and `:focus` styles.

Reply:
Well I'll leave this as an alternative for future planning